graph/auth.go

```diff
@@ -0,0 +1,23 @@
+package graph
+
+import (
+	"context"
+
+	"github.com/promoboxx/go-auth/src/auth"
+	"github.com/promoboxx/go-service/alice/middleware"
+)
+
+// This is a bit of a hack workaround to make the `middleware.GetClaimsFromContext` function return a pointer,
+// to gracefully handle situations where no auth claims were injected by any middleware. This change is necessary
+// because the GraphQL middleware does not prevent requests from reaching the resolvers if no token is present,
+// so they could panic with an unhelpful error message if they tried to call the standard function
+func EnsureAuthenticated(ctx context.Context) (c *auth.Claim) {
```

Comment:
Good use of recover and defer! So we can catch the panic of if we don't have a valid value from the context.

I think this would make more sense to error out if claims was nil then the caller needs to check the error value that might have been returned. The behavior of `!ensure_authenticated` in pbxx2 throws an error if the user is not authenticated or doesn't have a valid jwt. Semantically the word "ensure" to me would mean that this function is going to ensure that the ctx has a valid value to prove that the user is authenticated. Having to then check if the claims is nil outside of this function I think would be unexpected behavior.